app/controllers/category_controller.rb

```diff
@@ -3,22 +3,22 @@
 require 'warden'
 
 class CategoryController < ApplicationController
-  get '/create_category' do
-    redirect '/login' unless current_user
-    @current_user = current_user
+  get '/category/new' do
+    check_if_logged_in
+
     haml :'category/create_category'
   end
 
   get '/categories' do
-    redirect '/login' unless current_user
-    @current_user = current_user
+    check_if_logged_in
```

Comment:
I like that you've made this logic into a function. However, this is still a bit repetitive. It appears at the beginning of each route handler. Why not leverage **before filters**